crates/sui-core/src/authority/authority_store.rs

```diff
@@ -509,28 +510,13 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
             }
             ObjectLockStatus::LockedToTx { locked_by_tx } => locked_by_tx,
         };
-
-        // Returns None if either no TX with the lock, or TX present but no entry in transactions table.
-        // However we retry a couple times because the TX is written after the lock is acquired, so it might
-        // just be a race.
-        let tx_digest = &lock_info.tx_digest;
-        let mut retry_strategy = ExponentialBackoff::from_millis(2)
-            .factor(10)
-            .map(jitter)
-            .take(3);
-        let mut tx_option = self.epoch_tables().transactions.get(tx_digest)?;
-        while tx_option.is_none() {
-            if let Some(duration) = retry_strategy.next() {
-                // Wait to retry
-                tokio::time::sleep(duration).await;
-                trace!(?tx_digest, "Retrying getting pending transaction");
-            } else {
-                // No more retries, just quit
-                break;
-            }
-            tx_option = self.epoch_tables().transactions.get(tx_digest)?;
-        }
-        Ok(tx_option.map(|t| t.into()))
+        Ok(Some(
+            self.epoch_tables()
+                .transactions
+                .get(&lock_info.tx_digest)?
+                .ok_or_else(|| SuiError::GenericStorageError("Lock exists but cannot find transactions. Likely due to non-atomicity of the store".to_owned()))?
```

Comment:
are we sure we want to revert the workaround code? As I recall the workaround was added because we kept hitting this error.

Reply:
I guess I can leave it as it is. This code shouldn't be needed on validators, but I am not too sure about fullnodes.